### ISSUE TITLE
Accuracy by class – confusion_matrix_by_class()

### ISSUE THREAD

**Issue by matteozullo** [kaushaltrivedi/fast-bert#240]
Hi all,

I need to calculate accuracy by class in a multi-class classification problem, and I am using @Pawel-Kranzberg's confusion_matrix_by_class() function. How should I interpret the resulting confusion matrices?
In my toy example, I am testing BERT on a dataset with n=30 and I have three categories resulting in the following matrices:

array([[[ 0, 14],
        [ 0, 16]],

       [[24,  0],
        [ 6,  0]],

       [[22,  0],
        [ 8,  0]]])

I could not find any guidance for interpretation in prior issues.

Thank you

**Comment by Pawel-Kranzberg** [comment]
Hi @matteozullo.
the function is based on the scikit learn implementation of multilabel confusion matrices. So - paraphrasing the [relevant](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html) [docs](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.multilabel_confusion_matrix.html) -  each of your 2 x 2 matrices represent one of your three categories "as if binarized under a one-vs-rest transformation". In each matrix the "i-th row and j-th column entry indicates the number of samples with true label being i-th class and predicted label being j-th class", where class 0 is your given category and class 1 is the balance. Which basically means:
```
[[ True Positives, False Negatives], 
 [ False Positives, True Negatives]]
```

E.g., in your first matrix, 14 items from your first category were misclassified as not belonging to it (false negatives). In the second matrix, 24 items belonging to the second category were classified correctly (true positives). I take it you're actually dealing with a multi-label problem, not a multi-class one 😏
